### DISCUSSION TITLE
How to show a icon image in a Markdown note?

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#568] [Help] [answered]
Hi, 
I was trying to access the icons images to use them inside the notepanel, but I don't know if I'm doing something wrong or if I have to do some extra steps.

I was doing this simple test:

I have this node:
![image](https://user-images.githubusercontent.com/35700575/185246756-65ffaea4-fa74-48b7-bf62-27aecf96eba8.png)

and I added this formula in its note:
`=node.icons.urls`

So now its note show this result: `[bundle://1:1/images/icons/ksmiletris.svg] `

But when I try to use that url to show that image in a Markdown note, it doesn't work.

For example,
typing `![image](bundle://1:1/images/icons/ksmiletris.svg)` in a note and selecting content type Markdown doesn't do the trick.

this is what I get:
![image](https://user-images.githubusercontent.com/35700575/185248368-f5320666-6945-4ef1-8fa5-feb1a46324dd.png)

Is there a way to access that images?

Thanks in advance,
edo

**Comment by EdoFro** [comment]
I also tried to use HTML in the node's note to show the image:


```html
<html>
    <body>
        <img src="bundle://1:1/images/icons/ksmiletris.svg"
             alt="image"
             style="border-top-style: none; border-right-style: none; border-bottom-style: none; border-left-style: none"/>
    </body>
 </html>

```

But it doesn't work either 😞

**Comment by euu2021** [comment]
Did you try changing the slashes for backslashes in the link? I remember I had problems with slashes, in the past.

**Reply by EdoFro** [reply]
I think this is not the problem. It's an url: it should use slashes. 🤷🏼

**Comment by EdoFro** [comment, marked as answer]
Ok. I discovered!!
It's already there!!  in **HtmlUtils**!

## `htmlUtils.iconToHtml( Icon icon )`

Here is an example script. Select a node with an added icon and execute it:

```groovy
def nodo = node

def iconName = nodo.icons.first
def menuItemKey = 'IconAction.' + iconName
def icono = menuUtils.getMenuItemIcon(menuItemKey)
def html = htmlUtils.iconToHtml(icono)

def html2 =  "<html><body><p>icon ${html} is named '${iconName}'</p></body></html>"

def resultNode = nodo.createChild(html2)
resultNode.note = html2
resultNode.details = html2

```

**Reply by EdoFro** [reply]
here is an example image:
![image](https://user-images.githubusercontent.com/35700575/185643451-6cbcb81e-07b9-4718-8840-4d8b77ef260a.png)

**Reply by macmarrum** [reply]
A good find! :rocket:

**Comment by dpolivaev** [comment]
There is a way to include internal images in markdown and HTML but it works only with `png` or `jpg` bitmaps.

```
![image](freeplaneresource:/images/Freeplane_splash.png)
```
and 

```
<html>
  <head>
    
  </head>
  <body>
    <p>
      <img src="freeplaneresource:/images/Freeplane_splash.png"
             alt="image"/>
    </p>
  </body>
</html>
```
I think that both works only inside Freeplane but it is not supported by exports to other formats.

iconToHtml generates a bitmap and includes it as embedded data in format `data:image/png;base64,<imagedata>`.

**Reply by macmarrum** [reply]
Cool :+1:

**Reply by EdoFro** [reply]
Wow

And is there something similar to `freeplaneresource:` to access the **user directory**?

I want to add images to add-ons and show them into HTML or Markdown panels inside Freeplane (as part of a larger html text)

**Reply by EdoFro** [reply]
HEY!!!
I just discovered is the same  `freeplaneresource:`. It access the resources folder in the user directory too!!! 🎆 🍾 🥳

**Reply by dpolivaev** [reply]
The handler always checks if the resource is available under `<user-config-directory>/resources/` and uses it if it finds it there
